### PR TITLE
Leaner swarm/network and p2p/simulations

### PR DESCRIPTION
**Pull request by frncmx**
This PR is related to https://github.com/ethersphere/go-ethereum/issues/1059 where we want to move `node.go` from `swarm/network` to `p2p/simulations`.

This PR does not achieve that yet, but already became quite huge. So let's try to merge till I continue the move.

### CODE REVIEW

File: p2p/simulations/connect_test.go [hunk 1]
```diff
@@ -124,15 +124,30 @@ func TestConnectToRandomNode(t *testing.T) {
...
-	if err != nil {
-		t.Fatal(err)
+	tests := []struct {
+		name      string
+		nodeCount int
+	}{
+		{name: "no node", nodeCount: 0},
+		{name: "single node", nodeCount: 1},
+		{name: "2 nodes", nodeCount: 2},
+		{name: "3 nodes", nodeCount: 2},
```
Comment (frncmx) [inline p2p/simulations/connect_test.go]: 3 != 2

File: swarm/network_test.go [hunk 2]
```diff
@@ -374,6 +371,34 @@ func testSwarmNetwork(t *testing.T, o *testSwarmNetworkOptions, steps ...testSwa
 	}
 }
 
+func addNodesAndPreserveChainTopology(t *testing.T, s *simulation.Simulation, increment int) {
```
Comment (acud) [inline swarm/network_test.go]: don't need such a long function name. `addNodesChain` should suffice

File: swarm/network/simulation/node.go [hunk 3]
```diff
@@ -225,9 +119,6 @@ func (s *Simulation) UploadSnapshot(snapshotFile string, opts ...AddNodeOption)
 	for _, n := range snap.Nodes {
 		n.Node.Config.EnableMsgEvents = true
 		n.Node.Config.Services = s.serviceNames
-		for _, o := range opts {
```
Comment (acud) [inline swarm/network/simulation/node.go]: we still need this

File: swarm/network/simulation/node.go [hunk 4]
```diff
@@ -117,88 +89,10 @@ func (s *Simulation) AddNodes(count int, opts ...AddNodeOption) (ids []enode.ID,
...
-	if err != nil {
-		return nil, err
-	}
-	return ids, nil
-}
-
-// AddNodesAndConnectChain is a helpper method that combines
-// AddNodes and ConnectNodesChain. The chain will be continued from the last
-// added node, if there is one in simulation using ConnectToLastNode method.
-func (s *Simulation) AddNodesAndConnectChain(count int, opts ...AddNodeOption) (ids []enode.ID, err error) {
```
Comment (acud) [inline swarm/network/simulation/node.go:140]: the `AddNodeOption` should stay. we need to be able to control the configurations of the nodes that are being initialised

**Review by acud — CHANGES_REQUESTED** [review]

File: swarm/network/simulation/node.go (hunk 4 above)
Comment (frncmx) [reply swarm/network/simulation/node.go:140]: I've found no use case for that in the actual code. Do we have something in progress where we need that `opts` argument?

Comment (acud) [reply swarm/network/simulation/node.go:140]: https://github.com/ethersphere/go-ethereum/pull/1077#discussion_r243821759

Comment (frncmx) [reply swarm/network/simulation/node.go:140]: Oh, I've read that but I did not connect the two. Still, there we say "maybe in the future" and I would say [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) on that . If that is not too harsh from me. :)

Comment (frncmx) [reply swarm/network/simulation/node.go:140]: The commented code was removed eventually. However, `AddNodeOption` was brought back for `AddNode()` and `AddNodes()`.

**Review by zelig — COMMENTED** [review]
i disagree with this entire pr.
in general i wonder why we spending time on one of the best qualuty code in our code base.
typos and unexports aside i would revert all changes.

**Comment by acud** [comment]
In its current form me too. As mentioned before I disagree with removing the topology functions

**Comment by frncmx** [comment]
Ok. I'll cherry-pick the inspection fixes.

---

@justelad  and I had a little confusion around the concept of pivot, if I remember correctly. That concept is also used nowhere. But we still spent time on it and kinda duplicated the functionality when we moved `connect.go`. 

May I keep the commits removing the concept of pivot at least? Other commits I can let go.

**Pull request closed without merging**